python/cudf/cudf/core/column/string.py

```diff
@@ -5546,14 +5546,14 @@ def end_offset(self) -> int:
     @cached_property
     def memory_usage(self) -> int:
         n = 0
-        if len(self.base_children) == 2:
+        if len(self.base_children) == 1:
             child0_size = (self.size + 1) * self.base_children[
                 0
             ].dtype.itemsize
 
             child1_size = (
                 self.end_offset - self.start_offset
-            ) * self.base_children[1].dtype.itemsize
+            ) * cudf.api.types.dtype("int8").itemsize
```

Comment:
**issue*: This code makes no sense I think, there's one child and one `data` object. Why not:
```
return self.data.memory_usage() + self.base_children[0].memory_usage() + bitmask_usage
```
?

Reply:
In the older code itself, they don't include the entire second child's full memory. They calculate only the part of memory taken by start_offset and end_offset. 

Now, since we calculate `.data` properly for string column, the above suggestion will work. But `self.data` may incur a deep copy.

Reply:
Used `.data` for memory usage calculation